Fix Childhood Immunization Status map reduce infinite loop error

conroywhitney opened the pull request:
Duplicate `:medications` information was causing the `map_fn` for `Child Immunization Status (40280381-4555-E1C1-0145-D7C003364261)` to go into an infinite loop during mongo's mapreduce, eating up 100% CPU, and eventually causing a memory leak. Although the `BulkRecordImporter` calls `PatientImporter` which calls `Record`'s `dedup_record!` method, the `:medications` sections still were not being deduped. This fix uses ruby's `uniq` method to dedupe `:medications` rather than using more domain-specific approaches. This fixes the infinite loop error, and turns the corresponding Cypress tests green.

Commit:
Fix Childhood Immunization Status map reduce infinite loop error

Duplicate `:medications` information was causing the `map_fn` for `Child Immunization Status (40280381-4555-E1C1-0145-D7C003364261)` to go into an infinite loop during mongo's mapreduce, eating up 100% CPU, and eventually causing a memory leak. Although the `BulkRecordImporter` calls `PatientImporter` which calls `Record`'s `dedup_record!` method, the `:medications` sections still were not being deduped. This fix uses ruby's `uniq` method to dedupe `:medications` rather than using more domain-specific approaches. This fixes the infinite loop error, and turns the corresponding Cypress tests green.

dillonwelch commented:
Looks good to me. What do you think about making a PR for this on the main HDS repo?

conroywhitney commented:
To do that, should we make some sort of test for this in [record_test](https://github.com/q-centrix/health-data-standards/blob/master/test/unit/models/record_test.rb)?

conroywhitney commented:
^--- since our tests currently live in our main app

dillonwelch commented:
Yeah, we should probably do that anyway.

Commit:
Add test coverage for `dedup_section_naively!`

 * Rename `dedup_section_naively!`
 * Add fixture of duplicated medications patient
 * Add MiniTest to ensure `dudup_record!` works as expected

conroywhitney commented on lib/health-data-standards/models/record.rb:
```diff
@@ -121,13 +126,19 @@ def dedup_section_merging_codes_and_values!(section)
 
   def dedup_section!(section)
-    [:encounters, :procedures, :results].include?(section) ? dedup_section_merging_codes_and_values!(section) : dedup_section_ignoring_content!(section)
+    if [:encounters, :procedures, :results].include?(section)
```
Leaving this as is to minimize diff between pre-existing code.

Commit:
Load record fixtures in setup of record test

Continuous integration seems to have a problem finding the record we need to run `test_dedup_section_naively`. Adding loading of collection fixtures in `setup` method.

conroywhitney commented on test/unit/models/record_test.rb:
```diff
@@ -93,4 +93,15 @@ def test_dedup_record
\ No newline at end of file
+
+  def test_dedup_section_naively
+    # fixture: patient_with_duplicated_medications_section.json
```
fix'd (just not updating this snippet)
